rust/main/chains/hyperlane-ton/src/signer/signer.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Error;
+use base64::{engine::general_purpose, Engine};
+use hyperlane_core::ChainCommunicationError;
+use tonlib_core::{
+    cell::{ArcCell, BagOfCells, Cell},
+    mnemonic::{KeyPair, Mnemonic},
+    wallet::{TonWallet, WalletVersion},
+    TonAddress,
+};
+
+#[derive(Clone)]
+pub struct TonSigner {
+    pub address: TonAddress,
+    pub wallet: TonWallet,
+}
+
+impl TonSigner {
+    pub fn new(key_pair: KeyPair, wallet_version: WalletVersion) -> Result<Self, Error> {
+        let wallet =
+            TonWallet::derive_default(wallet_version, &key_pair).map_err(|e| Error::new(e))?;
+
+        Ok(TonSigner {
+            address: wallet.address.clone(),
+            wallet,
+        })
+    }
+    pub fn from_mnemonic(
```

Comment:
A new line before next method definition:
```suggestion
    }

    pub fn from_mnemonic(
```

Reply:
Ok, fixed